zjit/src/backend/x86_64/mod.rs

```diff
@@ -476,22 +477,34 @@ impl Assembler
                     cb.write_byte(0);
                 },
 
-                // Set up RBP to work with frame pointer unwinding
+                // Set up RBP as frame pointer work with unwinding
                 // (e.g. with Linux `perf record --call-graph fp`)
-                Insn::FrameSetup => {
-                    if false { // We don't support --zjit-perf yet
-                        // TODO(alan): Change Assembler::frame_size() when adding --zjit-perf support
-                        push(cb, RBP);
-                        mov(cb, RBP, RSP);
-                        push(cb, RBP);
+                // and to allow push and pops in the function.
+                &Insn::FrameSetup { preserved, mut slot_count } => {
```

Comment:
To make it a bit more consistent with `total_slots`, I'm tempted to call it `num_slots` instead of `slot_count`.

Reply:
Ayy unless you feel strongly about this I'm going to keep it as is since I prefer noun adjective form for public-ish things like function names and field names.